fdbctl/protos/control_service.proto

```diff
@@ -2,5 +2,399 @@ syntax = "proto3";
 
 package fdbctl;
 
+option go_package = "github.com/apple/foundationdb/fdbctl";
+option java_package = "com.apple.foundationdb.fdbctl";
+
+//------ RPCs ------
+
+// ControlService provides administrative and management operations for a FoundationDB cluster.
+// This service allows control over cluster configuration, coordinator management, worker lifecycle,
+// and monitoring of cluster health and status.
 service ControlService {
+    // Retrieves the current list of coordinators for the cluster.
+    rpc GetCoordinators(GetCoordinatorsRequest) returns (GetCoordinatorsReply);
+
+    // Changes the cluster's coordinators. This is a critical operation that should be
+    // performed carefully as coordinators maintain cluster configuration state.
+    rpc ChangeCoordinators(ChangeCoordinatorsRequest) returns (ChangeCoordinatorsReply);
+
+    // Suggests optimal configuration parameters based on current cluster topology.
+    rpc ConfigureAutoSuggest(ConfigureAutoSuggestRequest) returns (ConfigureAutoSuggestReply);
+
+    // Configures database settings including redundancy mode, storage engine, process counts,
+    // encryption, and other critical cluster parameters.
+    rpc Configure(ConfigureRequest) returns (ConfigureReply);
+
+    // Retrieves the current read version (transaction version) of the database.
+    rpc GetReadVersion(GetReadVersionRequest) returns (GetReadVersionReply);
+
+    // Retrieves comprehensive cluster status including health, performance, and configuration.
+    rpc GetStatus(GetStatusRequest) returns (GetStatusReply);
+
+    // Retrieves the list of all worker processes in the cluster.
+    rpc GetWorkers(GetWorkersRequest) returns (GetWorkersReply);
+
+    // Re-enables previously excluded workers, allowing them to rejoin the cluster.
+    rpc Include(IncludeRequest) returns (IncludeReply);
+
+    // Excludes workers from the cluster, triggering graceful data migration away from them.
+    rpc Exclude(ExcludeRequest) returns (ExcludeReply);
+
+    // Retrieves the status of worker exclusions including in-progress migrations.
+    rpc ExcludeStatus(ExcludeStatusRequest) returns (ExcludeStatusReply);
+
+    // Forcefully terminates worker processes. Use with caution - prefer Exclude for graceful removal.
+    rpc Kill(KillRequest) returns (KillReply);
+}
+
+//------ Messages -------
+
+// Worker represents a FoundationDB process in the cluster.
+// Workers can have different roles (storage, transaction, etc.) and are
+// identified by their network address and locality information.
+message Worker {
+    // Locality describes the physical or logical location of a worker process.
+    // These fields help FoundationDB make intelligent placement decisions for
+    // data replication and fault tolerance.
+    message Locality {
```

Comment:
Additional locality fields will be ignored in the `Locality` message, correct? Might be worth to note that in the comments.